docs/using-ngrok-with/k8s.md

```diff
@@ -0,0 +1,246 @@
+------------
+
+## Introduction
+
+The [ngrok Ingress Controller for Kubernetes](https://github.com/ngrok/kubernetes-ingress-controller) is our official open-source controller for adding public and secure ingress traffic to your k8s clusters. If you’ve used ngrok in the past, you can think of the ngrok Ingress Controller as ngrok packaged an idiomatic k8s resource — available as helm chart, configurable via manifest (using the `kind: Ingress` construct), and compatible with k8s best practices. 
+
+In this tutorial, you will install the ngrok Ingress Controller and run a sample 2048 app with public ingress access and security provided by ngrok.
+
+:::caution This tutorial requires:
+1. A local environment with Kubernetes installed and configured. For this tutorial, we will use `k3d`, `kubectl`, and `helm`.
+1. An [ngrok account](https://ngrok.com/signup) with a Pro subscription. The subscription is required in order to configure the ngrok Ingress Controller with a custom domain and an Edge Configuration.
+:::
+
+## Get started with the ngrok Ingress Controller for Kubernetes
+
+## Step 1: Get your ngrok authtoken and API key
+
+:::tip
+The ngrok Ingress Controller requires:
+- The authtoken to launch connections from ngrok's points of presence to your k8s cluster.
+- The API key to set ngrok edge configurations for your k8s cluster
+:::
+
+To started with the ngrok Ingress Controller for Kubernetes:
+
+1. Access the [ngrok Dashboard](https://dashboard.ngrok.com) with your Pro account.
+1. Click [Your Authtoken](https://dashboard.ngrok.com/get-started/your-authtoken). Copy the Authtoken to a text editor.
+1. Click [API](https://dashboard.ngrok.com/api).
+1. Follow the instructions to create a new API key. Copy the API key to a text editor.
+
+## Step 2: Setup your Kubernetes cluster and install the ngrok Ingress Controller
+
+1. In your terminal, create a k8s cluster:
+    
+    ```bash
+    k3d cluster create ngrok
+    ```
+
+1. Using `helm`, add the ngrok repo:
+
+    ```bash
+    helm repo add ngrok https://ngrok.github.io/kubernetes-ingress-controller
+    ```
+
+1. Install the ngrok Ingress Controller in your cluster, replacing `[AUTHTOKEN]` and `[API_KEY]` with your Authtoken and API key from above:
+
+    ```bash
+    export NGROK_AUTHTOKEN=[AUTHTOKEN]
+    export NGROK_API_KEY=[API_KEY]
+    helm install ngrok-ingress-controller ngrok/kubernetes-ingress-controller \
+    --set credentials.apiKey=$NGROK_API_KEY \
+    --set credentials.authtoken=$NGROK_AUTHTOKEN
+    kubectl apply -f ngrok-manifest.yaml
+    ```
+
+1. Create a manifest file (for example `ngrok-manifest.yaml`) with the following contents:
+
+    :::tip Notes:
+    - Lines 1-32: Create the 2048 app service and deployment
+    - Lines 34-51: Create the ngrok Ingress Controller. Line 42 in specific determines the ingress URL for public requests.
+    :::
+
+    ```yaml showLineNumbers
+    apiVersion: v1
+    kind: Service
+    metadata:
+    name: game-2048
```

Comment:
one thing that might help people avoid namespace problems like you had would be if every resource specifies the namespace, otherwise it uses whatever namespace they have set in their context. 

Everything not having a namespace should make them all end up in the same namepace asuming someone doesn't change namespace halfway through, but if they do and the manifets have namespaces set, they'll take precedence.

Reply:
added namespace for all resources. Also brought back create a namespace to the helm install command